File: solutions/1603-running-sum-of-1d-array/solution.java
class Solution {
    public int[] runningSum(int[] nums) {
        int arr[]=new int[nums.length];
        int sum=0,i=0;
        for(int n:nums)
        {
            sum+=n;
            arr[i++]=sum;
        }
        return arr;
    }
}

File: solutions/1603-running-sum-of-1d-array/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* runningSum(int* nums, int numsSize, int* returnSize) {
    *returnSize=numsSize;
    int adder=0;
    for(int i=0;i<numsSize;i++)
    {
        adder+=nums[i];
        nums[i]=adder;
    }
    return nums;
}
